lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -779,9 +779,27 @@ def upload_grades_csv(_xmodule_instance_args, _entry_id, course_id, _task_input,
 
         if not course_grade:
             # An empty gradeset means we failed to grade a student.
-            task_progress.failed += 1
-            err_rows.append([student.id, student.username, err_msg])
-            continue
+            # TODO: create a separate track report so that instructors
+            # do not need to use grade reports to view learners' track
+            # before the course starts. We special-case for this scenario
+            # here... for now.
+            if err_msg == "User does not have access to this course" and (
+                            course.start > datetime.now().replace(tzinfo=UTC) or not course.start is None
+            ):
```

Comment:
I think it makes sense to improve the error handling within the `iter` method in this case.  The expectation of the `iter` method is that it iterates through all the users and returns their grades.  You should then add an additional value in the `GradeResult` enum to indicate that the user doesn't have access to the course.